types/v56/ip_space.go

```diff
@@ -200,3 +200,119 @@ type IPPrefixSequence struct {
 	// Specifies the percentage of allocated IP prefix blocks out of total specified IP prefix blocks.
 	AllocatedPrefixPercentage float32 `json:"allocatedPrefixPercentage,omitempty"`
 }
+
+// IpSpaceUplink sSpecifies the IP Space Uplink configuration for Provider Gateway (External network
+// with T0 or T0 VRF backing)
+type IpSpaceUplink struct {
+	ID          string `json:"id,omitempty"`
+	Name        string `json:"name"`
+	Description string `json:"description,omitempty"`
+	// ExternalNetworkRef contains information
+	ExternalNetworkRef *OpenApiReference `json:"externalNetworkRef"`
+	IPSpaceRef         *OpenApiReference `json:"ipSpaceRef"`
+	// The type of the IP Space associated with this uplink. Possible values are: PUBLIC, PRIVATE,
+	// SHARED_SERVICES. This property is read-only.
+	IPSpaceType string `json:"ipSpaceType,omitempty"`
+	Status      string `json:"status,omitempty"`
+}
+
+// IpSpaceIpAllocationRequest is an IP Space IP Allocation request object. An IP Space IP allocation request can
+// either request a specific IP address/IP prefix or request a specific number of any free IP
+// Addresses/IP Prefixes within an IP Space. To allocate a specific IP Address or IP Prefix, the
+// value field should be used and the IP Address or Prefix should be specified. To allocate up to 5
```

Comment:
Could you explain a bit this limitation? Is it because the API doesn't return more than 5?

Reply:
Yes. But now as you reiterate that - I am not sure if I should even note that - this is a thing that can change with versions.

Reply:
I removed this particular limitation, as it might age too quick